File: BaseLib/src/main/java/com/hjc/baselib/utils/helper/ActivityManager.java
package com.hjc.baselib.utils.helper;

import android.app.Activity;

import java.util.LinkedList;
import java.util.List;

/**
 * @Author: HJC
 * @Date: 2019/1/7 11:37
 * @Description: Activity管理类
 */
public class ActivityManager {
    private static List<Activity> activityList = new LinkedList<>();

    /**
     * 添加指定的Activity
     */
    public static void addActivity(Activity activity) {
        if (!activityList.contains(activity)) {
            activityList.add(activity);
        }
    }

    /**
     * 销毁所有的Activity
     */
    public static void finishAllActivities() {
        for (Activity activity : activityList) {
            activity.finish();
        }
    }

    /**
     * 销毁指定的Activity
     */
    public static void removeActivity(Class<?> cls) {
        Activity tempActivity = null;
        for (Activity activity : activityList) {
            if (activity.getClass().equals(cls)) {
                tempActivity = activity;
            }
        }

        finishSingleActivity(tempActivity);
    }

    /**
     * 结束指定的Activity
     */
    private static void finishSingleActivity(Activity activity) {
        if (activity != null) {
            if (activityList.contains(activity)) {
                activityList.remove(activity);
            }
            activity.finish();
        }
    }
}

File: BaseLib/src/main/java/com/hjc/baselib/utils/helper/ActivityManager.kt
package com.hjc.baselib.utils.helper

import android.app.Activity
import java.util.*

/**
 * @Author: HJC
 * @Date: 2019/1/7 11:37
 * @Description: Activity管理类
 */
object ActivityManager {
    private val activityList: MutableList<Activity> = LinkedList()

    /**
     * 添加指定的Activity
     */
    fun addActivity(activity: Activity) {
        if (!activityList.contains(activity)) {
            activityList.add(activity)
        }
    }

    /**
     * 销毁所有的Activity
     */
    fun finishAllActivities() {
        for (activity in activityList) {
            activity.finish()
        }
    }

    /**
     * 销毁指定的Activity
     */
    fun removeActivity(cls: Class<*>) {
        var tempActivity: Activity? = null
        for (activity in activityList) {
            if (activity.javaClass == cls) {
                tempActivity = activity
            }
        }
        finishSingleActivity(tempActivity)
    }

    /**
     * 结束指定的Activity
     */
    private fun finishSingleActivity(activity: Activity?) {
        if (activity != null) {
            if (activityList.contains(activity)) {
                activityList.remove(activity)
            }
            activity.finish()
        }
    }
}